<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ./urdf/xacro/rt_model_fk.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="touch_right">
  <material name="black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="gray">
    <color rgba="0.2 0.2 0.2 1.0"/>
  </material>
  <material name="silver">
    <color rgba="0.9137254901960784 0.9137254901960784 0.8470588235294118 1.0"/>
  </material>
  <material name="orange">
    <color rgba="1.0 0.4235294117647059 0.0392156862745098 1.0"/>
  </material>
  <material name="brown">
    <color rgba="0.8705882352941177 0.8117647058823529 0.7647058823529411 1.0"/>
  </material>
  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="white">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <link name="base_link">
        
    </link>
  <link name="foot_right">
    <visual>
      <origin rpy="0 0 0" xyz="-0.12 0 0.025"/>
      <geometry>
        <box size="0.24 0.1 0.05"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.12 0 0.025"/>
      <geometry>
        <box size="0.12 0.05 0.025"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.9179"/>
      <origin rpy="0 0 0" xyz="-0.13404 0 0.025"/>
      <inertia ixx="0.0008129451110399999" ixy="0" ixz="0" iyy="0.003173584176" iyz="0" izz="0.00349207932096"/>
    </inertial>
  </link>
  <joint name="toe" type="revolute">
    <parent link="base_link"/>
    <child link="foot_right"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" lower="-3.14" upper="3.14" velocity="21"/>
  </joint>
  <link name="shank_right">
    <visual>
      <origin rpy="0 0 0" xyz=" 0 0 0.2"/>
      <geometry>
        <box size="0.1 0.1 0.4"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.2"/>
      <geometry>
        <box size="0.05 0.05 0.2"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2.9011"/>
      <origin rpy="0 0 0" xyz="0 0 0.22164000000000003"/>
      <inertia ixx="0.030183044400000004" ixy="0" ixz="0" iyy="0.028779376176" iyz="0" izz="0.004924443184"/>
    </inertial>
  </link>
  <joint name="ankle_right" type="revolute">
    <origin rpy="0 0 0" xyz="-0.24 0 0.05"/>
    <parent link="foot_right"/>
    <child link="shank_right"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" lower="-0.78" upper="0.78" velocity="21"/>
  </joint>
  <link name="thigh_right">
    <visual>
      <origin rpy="0 0 0" xyz=" 0 0 0.2"/>
      <geometry>
        <box size="0.1 0.1 0.4"/>
      </geometry>
      <material name="red"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.2"/>
      <geometry>
        <box size="0.05 0.05 0.2"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="9.4872"/>
      <origin rpy="0 0 0" xyz="0 0 0.23620000000000002"/>
      <inertia ixx="0.16430464243200002" ixy="0" ixz="0" iyy="0.16430464243200002" iyz="0" izz="0.033700052352"/>
    </inertial>
  </link>
  <joint name="knee_right" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.4"/>
    <parent link="shank_right"/>
    <child link="thigh_right"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" lower="-2.35" upper="0" velocity="21"/>
  </joint>
  <link name="trunk">
    <visual>
      <origin rpy="0 0 0" xyz=" 0 0.15 0.35"/>
      <geometry>
        <box size="0.2 0.35 0.7"/>
      </geometry>
      <material name="gray"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.15 0.35"/>
      <geometry>
        <box size="0.1 0.175 0.35"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="29.118199999999998"/>
      <origin rpy="0 0 0" xyz="0 0.15 0.38598"/>
      <inertia ixx="1.9744515645119995" ixy="0" ixz="0" iyy="1.7179857384619994" iyz="0" izz="0.5205079165579999"/>
    </inertial>
  </link>
  <joint name="hip_right" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.4"/>
    <parent link="thigh_right"/>
    <child link="trunk"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" lower="-0.523" upper="1.57" velocity="21"/>
  </joint>
  <link name="thigh_left">
    <visual>
      <origin rpy="0 0 0" xyz=" 0 0 -0.2"/>
      <geometry>
        <box size="0.1 0.1 0.4"/>
      </geometry>
      <material name="red"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.2"/>
      <geometry>
        <box size="0.05 0.05 0.2"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="9.4872"/>
      <origin rpy="0 0 0" xyz="0 0 -0.1638"/>
      <inertia ixx="0.16430464243200002" ixy="0" ixz="0" iyy="0.16430464243200002" iyz="0" izz="0.033700052352"/>
    </inertial>
  </link>
  <joint name="hip_left" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.3 0"/>
    <parent link="trunk"/>
    <child link="thigh_left"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" lower="-1.57" upper="0.523" velocity="21"/>
  </joint>
  <link name="shank_left">
    <visual>
      <origin rpy="0 0 0" xyz=" 0 0 -0.2"/>
      <geometry>
        <box size="0.1 0.1 0.4"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.2"/>
      <geometry>
        <box size="0.05 0.05 0.2"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2.9011"/>
      <origin rpy="0 0 0" xyz="0 0 -0.17836000000000002"/>
      <inertia ixx="0.030183044400000004" ixy="0" ixz="0" iyy="0.028779376176" iyz="0" izz="0.004924443184"/>
    </inertial>
  </link>
  <joint name="knee_left" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.4"/>
    <parent link="thigh_left"/>
    <child link="shank_left"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" lower="0" upper="2.35" velocity="21"/>
  </joint>
  <link name="foot_left">
    <visual>
      <origin rpy="0 0 0" xyz="0.12 0 -0.025"/>
      <geometry>
        <box size="0.24 0.1 0.05"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.12 0 -0.025"/>
      <geometry>
        <box size="0.12 0.05 0.025"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.9179"/>
      <origin rpy="0 0 0" xyz="0.10596 0 -0.025"/>
      <inertia ixx="0.0008129451110399999" ixy="0" ixz="0" iyy="0.003173584176" iyz="0" izz="0.00349207932096"/>
    </inertial>
  </link>
  <joint name="ankle_left" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.4"/>
    <parent link="shank_left"/>
    <child link="foot_left"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" lower="-0.78" upper="0.78" velocity="21"/>
  </joint>
  <link name="foot_right_com">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="foot_right_com_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.13404 0 0.025"/>
    <parent link="foot_right"/>
    <child link="foot_right_com"/>
  </joint>
  <link name="shank_right_com">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="shank_right_com_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.22164000000000003"/>
    <parent link="shank_right"/>
    <child link="shank_right_com"/>
  </joint>
  <link name="thigh_right_com">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="thigh_right_com_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.23620000000000002"/>
    <parent link="thigh_right"/>
    <child link="thigh_right_com"/>
  </joint>
  <link name="trunk_inertia">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.002"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="trunk_inertia_fixed" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.15 0.35"/>
    <parent link="trunk"/>
    <child link="trunk_inertia"/>
  </joint>
  <link name="trunk_com">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="trunk_com_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.15 0.38598"/>
    <parent link="trunk"/>
    <child link="trunk_com"/>
  </joint>
  <link name="thigh_left_com">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="thigh_left_com_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.1638"/>
    <parent link="thigh_left"/>
    <child link="thigh_left_com"/>
  </joint>
  <link name="shank_left_com">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="shank_left_com_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.1768"/>
    <parent link="shank_left"/>
    <child link="shank_left_com"/>
  </joint>
  <link name="foot_left_com">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="foot_left_com_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.10596 0 -0.025"/>
    <parent link="foot_left"/>
    <child link="foot_left_com"/>
  </joint>
  <link name="foot_right_contact0">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
      <material name="orange"/>
    </visual>
  </link>
  <link name="foot_right_contact1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
      <material name="orange"/>
    </visual>
  </link>
  <link name="foot_right_contact2">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
      <material name="orange"/>
    </visual>
  </link>
  <link name="foot_right_contact3">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
      <material name="orange"/>
    </visual>
  </link>
  <joint name="foot_right_contact0_fixed" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0.05 0"/>
    <parent link="foot_right"/>
    <child link="foot_right_contact0"/>
  </joint>
  <joint name="foot_right_contact1_fixed" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 -0.05 0"/>
    <parent link="foot_right"/>
    <child link="foot_right_contact1"/>
  </joint>
  <joint name="foot_right_contact2_fixed" type="fixed">
    <origin rpy="0 0 0" xyz="-0.24 -0.05 0"/>
    <parent link="foot_right"/>
    <child link="foot_right_contact2"/>
  </joint>
  <joint name="foot_right_contact3_fixed" type="fixed">
    <origin rpy="0 0 0" xyz="-0.24 0.05 0"/>
    <parent link="foot_right"/>
    <child link="foot_right_contact3"/>
  </joint>
  <link name="foot_left_contact0">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
      <material name="orange"/>
    </visual>
  </link>
  <link name="foot_left_contact1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
      <material name="orange"/>
    </visual>
  </link>
  <link name="foot_left_contact2">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
      <material name="orange"/>
    </visual>
  </link>
  <link name="foot_left_contact3">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
      <material name="orange"/>
    </visual>
  </link>
  <joint name="foot_left_contact0_fixed" type="fixed">
    <origin rpy="0 0 0" xyz="0.24 0.05 -0.05"/>
    <parent link="foot_left"/>
    <child link="foot_left_contact0"/>
  </joint>
  <joint name="foot_left_contact1_fixed" type="fixed">
    <origin rpy="0 0 0" xyz="0.24 -0.05 -0.05"/>
    <parent link="foot_left"/>
    <child link="foot_left_contact1"/>
  </joint>
  <joint name="foot_left_contact2_fixed" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 -0.05 -0.05"/>
    <parent link="foot_left"/>
    <child link="foot_left_contact2"/>
  </joint>
  <joint name="foot_left_contact3_fixed" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0.05 -0.05"/>
    <parent link="foot_left"/>
    <child link="foot_left_contact3"/>
  </joint>
</robot>

